lib/api/controllers/adminController.js

```diff
@@ -54,6 +58,15 @@ class AdminController {
         return this.kuzzle.internalEngine.bootstrap.all();
       });
 
+    if (refresh === 'wait_for') {
+      return promise.then(() => {
+        delete _locks.resetKuzzleData;
+
+        return { acknowledge: true };
+      });
+    }
+
+    delete _locks.resetKuzzleData;
```

Comment:
I disagree: this lock should be removed at the end of the reset, even if the API do respond asynchronously to the requesting user.